src/Symfony/Component/HttpFoundation/Session/Storage/Handler/MigratingSessionHandler.php

```diff
@@ -22,8 +22,8 @@
  */
 class MigratingSessionHandler implements \SessionHandlerInterface, \SessionUpdateTimestampHandlerInterface
 {
-    private $currentHandler;
-    private $writeOnlyHandler;
+    private \SessionHandlerInterface|\SessionUpdateTimestampHandlerInterface $currentHandler;
+    private \SessionHandlerInterface|\SessionUpdateTimestampHandlerInterface $writeOnlyHandler;
```

Comment:
IIUC, an intersection would be better, if possible,isn't it?

Reply:
in the meantime, I suggest using `\SessionHandlerInterface` instead of a union then (and maybe use phpdoc to document the actual intersection type) instead of using a union type that make no sense here.